nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -177,8 +161,10 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
                 )
         if hasattr(dataset, 'collate_fn'):
             collate_fn = dataset.collate_fn
-        else:
+        elif hasattr(dataset.datasets[0], 'collate_fn'):
             collate_fn = dataset.datasets[0].collate_fn
+        else:
+            collate_fn = dataset.datasets[0].datasets[0].collate_fn
```

Comment:
Lol what's happening here

Reply:
We need much more docstring to explain this

Reply:
added comments. This is to support concat datasets that are a collection of tarred datasets (that are a collection of tar files)